types/inquirer/index.d.ts

```diff
@@ -70,90 +72,158 @@ declare namespace inquirer {
         restoreDefaultPrompts(): void;
     }
 
-    interface Question<T = Answers> {
+    interface QuestionCommon<A> {
         /**
-         * Type of the prompt.
-         * Possible values:
-         * <ul>
-         *      <li>input</li>
-         *      <li>number</li>
-         *      <li>confirm</li>
-         *      <li>list</li>
-         *      <li>rawlist</li>
-         *      <li>password</li>
-         * </ul>
-         * @defaults: 'input'
+         * The name to use when storing the answer in the answers hash.
+         * If the name contains periods, it will define a path in the answers hash.
          */
-        type?: string;
```

Comment:
I'm sorry for previous comment [at a wrong position](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/35489/files#r285837466). Actually, what I did not understand is why not keep `type: string`  as a placeholder here and then the specific `type`s given in each detailed question interfaces could be viewed as override(or implement) this `type` property. Because `type` property is exactly shared by all question types excepting the values are various but all are strings.

Reply:
In inquirer you can optionally add and use custom question-types.
Having this in mind a type-property in the base class seems reasonable to me

What do you guys think?

Reply:
I think if you have custom question types then you should define them in a type, just as yeoman-generator ads a `store` prop. If you want something more loose than that, then you can define a looser type too. I'll give this some more thought and look for documentation, because it's nowhere in the examples.

Reply:
@manuth it looks like someone tried to introduce custom prompt types, but the issue and PR were closed. See https://github.com/SBoudrias/Inquirer.js/pull/238

Reply:
Sorry - my bad
What I was talking about was custom prompts:
https://github.com/SBoudrias/Inquirer.js/blob/master/README.md#inquirerregisterpromptname-prompt

People can create their own prompts and they are accessible by setting their `name` in a question's `type`.

Reply:
Also, fundamentally, I don't think this suggestion offers anything at all, because it's overridden in each specific question type. Take this TS for example:
```ts
interface Base {
	type: string
}

interface Foo extends Base {
	type: 'foo'
}

const subject: Foo = { type: 'bar' }

// Type '"bar"' is not assignable to type '"foo"'.
// (property) Foo.type: "foo"
```
Because the `Base` interface is only referenced by the specific question types and not directly.

Reply:
I totally agree - let's wait for @chigix's feedback.

Reply:
@manuth, I would refer back to [this comment](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/35489#discussion_r286014246) concerning what to do about that particular use case. People can pass-in anything, so long as it extends `inquirer.Answers`. Gotta' love generics!

Reply:
I hope I get you right - my English's pretty messy sometimes.
On one hand declaring `type: string` gives people writing their own prompts a plan on what to do about the `type`-property (which is: declaring an own `name` to access it using the `Question`s `type`-property).

On the other hand people could set a `type` that doesn't even exist as long as they're using the `QuestionCommon<A>`-interface rather than some more specific interface.

Though... to me the `QuestionCommon<A>` interface _must_ have a `type`-member as you _must_ set the `type` no matter what kind of prompt you're trying to show, right...?

Reply:
No, actually. The `type` [defaults to "input"](https://github.com/SBoudrias/Inquirer.js#question), which why I have it written this way: 

```ts
    interface InputQuestion<A> extends QuestionCommon<A>,
        Pick<QuestionOptions<A>, 'filter' | 'transformer'> {
        type?: 'input'
```
This is the only question type that has an optional `type` prop, because if the user doesn't supply a `type` an `InputQuestion` is assumed. It's not a required prop, but it does have a default value of "input".

Defining these specific types prevents the developer from making mistakes like:
1. Using a type that doesn't exist.
1. Adding props that are not supported for the specified type.

The current type definitions for Inquirer are too loose and do not prevent these things from happening, which is why I thought it was important to fix in this PR.

Reply:
Aaah - so now I get your point.
Do you think there is any other change required in order to make the usage of `inquirer` more smoothly?

Reply:
Yes, I do, but this PR is already a bit much and I'd like to seal it off at this point. I might be adding another (much smaller) PR in the near future.

But yes, I've run into some hiccups during my Yeoman generator development that could be much smoother.